docs/src/snoop_pc.md

```diff
@@ -0,0 +1,70 @@
+# SnoopPrecompile
+
+SnoopPrecompile provides a macro, `@precompile_calls`, which precompiles every call on its first usage.
+The key feature of `@precompile_calls` is that it intercepts all callees (including those made by runtime dispatch) and, on Julia 1.8 and higher, allows you to precompile the callee even if it comes from a different module (e.g., `Base` or a different package).
+
+Statements that occur inside a `@precompile_calls` block are executed only if the package is being actively precompiled;
+it does not run when the package is loaded, nor if you're running Julia with `--compiled-modules=no`.
+
+Here's an illustration of how you might use `@precompile_calls`:
+
+```julia
+module MyPackage
+
+using SnoopPrecompile    # this is a small dependency
+
+struct MyType
+    x::Int
+end
+struct OtherType
+    str::String
+end
+
+let  # `let` prevents `list` from being a visible private name in the package
+    @precompile_calls :setup beginNaturally, y
```

Comment:
What is `beginNaturally, y` here?

Reply:
Weird vscode paste! Should just be `begin`. Every once in a while vscode does weird things to my text, I must be sloppy about exiting multicursor or something?